src/components/DragAndDropExample.js

```diff
@@ -0,0 +1,280 @@
+        null,
+        {
+          dx: this.state.pan[index].x,
+          dy: this.state.pan[index].y
+        }
+      ]),
+      onPanResponderRelease: (e, gesture) => {
+        this.state.pan[index].flattenOffset();
+        const springTo = {
+          x: this.initialPositions[index].x,
+          y: this.initialPositions[index].y
+        };
+        const itemLinkedZone = this.state.itemLinkedZone;
+        let linkedZoneFound = false;
+        for (let i = 0; i < this.zoneLayouts.length; i++) {
+          const zone = this.zoneLayouts[i];
+          if (
+            zone.x < gesture.moveX &&
+            gesture.moveX < zone.x + zone.width &&
+            zone.y < gesture.moveY &&
+            gesture.moveY < zone.y + zone.height
+          ) {
+            springTo.x =
+              zone.x + zone.width / 2 - this.itemLayouts[index].width / 2;
+            springTo.y =
+              zone.y + zone.height / 2 - this.itemLayouts[index].height / 2;
+            linkedZoneFound = true;
+            secondItemInZone = this.state.itemLinkedZone.indexOf(i);
+            if (secondItemInZone >= 0) {
+              itemLinkedZone[secondItemInZone] = itemLinkedZone[index];
+              if (itemLinkedZone[index] === -1) {
+                Animated.spring(this.state.pan[secondItemInZone], {
+                  toValue: {
+                    x: this.initialPositions[secondItemInZone].x,
+                    y: this.initialPositions[secondItemInZone].y
+                  }
+                }).start();
+              }
+            }
+            itemLinkedZone[index] = i;
+            this.setState({ itemLinkedZone });
+            break;
+          }
+        }
+        if (!linkedZoneFound) {
+          itemLinkedZone[index] = -1;
+          this.setState({ itemLinkedZone });
+        }
+        Animated.spring(this.state.pan[index], {
+          toValue: { ...springTo }
+        }).start();
+        this.setState({ movingItem: -1 });
+      }
+    });
+  };
+
+  updateZoneLayout = (event, index) => {
+    const layout = event.nativeEvent.layout;
+    this.zoneLayouts[index] = layout;
+    const itemIndex = this.state.itemLinkedZone.indexOf(index);
+    if (itemIndex >= 0) {
+      const springTo = {
+        x: layout.x + layout.width / 2 - this.itemLayouts[itemIndex].width / 2,
+        y: layout.y + layout.height / 2 - this.itemLayouts[itemIndex].height / 2
+      };
+      Animated.spring(this.state.pan[itemIndex], {
+        toValue: { ...springTo }
+      }).start();
+    }
+  };
+
+  renderZones = n => {
+    const zones = [];
+    for (let i = 0; i < n; i++) {
+      const itemIndex = this.state.itemLinkedZone.indexOf(i);
+      zones.push(
+        <View
+          key={`zone_${i}`}
+          onLayout={e => this.updateZoneLayout(e, i)}
+          style={{
+            justifyContent: 'center',
+            alignItems: 'center',
+            width: itemIndex >= 0 ? this.itemLayouts[itemIndex].width : 50,
```

Comment:
If a **word** (item) doesn't belong to any **zone** (not dragged there) - zone should have some default width or maybe I'm overthinking here and there is a simpler solution? :)